### PR TITLE
Feat/swager

### PR DESCRIPTION
**Pull request by sarkis1231**
Screenshot 2024-05-23 at 9 02 51 PM

### CODE REVIEW

**Comment by Mgrdich** [comment]
let's not merge this until we merge tests suites into main first @sarkis1231

File: internal/routes/routes.go
Comment (Mgrdich) [inline internal/routes/routes.go]: this should only be mounted in the dev mode.

**Review by Mgrdich — COMMENTED** [review]
we need to discuss if we really need to push the swagger `json` and `yaml` files , since they are auto generated i think  everyone can do it for themselves, i think they can be added in `gitignore` file. same goes for the docs.go file as well, everything that is auto generated.

what do you think ?

**Comment by sarkis1231** [comment]
> we need to discuss if we really need to push the swagger `json` and `yaml` files , since they are auto generated i think everyone can do it for themselves, i think they can be added in `gitignore` file. same goes for the docs.go file as well, everything that is auto generated.
> 
> what do you think ?

Yes since we are going to use it just for dev env we can add those files to gitignore.
The reason that kept the files there is that I'm importing docs in the server.go

File: .github/workflows/testGoApi.yml
Comment (Mgrdich) [inline .github/workflows/testGoApi.yml]: this needs to have a specific value , cause we might run locally different version and due to an update we can encounter an error that we did not face in the dev.

otherwise everything looks good.

**Review by Mgrdich — APPROVED** [review]

**Pull request merged**
